lib/src/rules/comment_references.dart

```diff
@@ -49,10 +50,42 @@ class Visitor extends SimpleAstVisitor {
   Visitor(this.rule);
 
   @override
+  visitComment(Comment node) {
+    // Check for keywords that are not treated as references by the parser
+    // but should be flagged by the linter.
+    // Note that no special care is taken to handle embedded code blocks.
+    for (Token token in node.tokens) {
+      if (!token.isSynthetic) {
+        String comment = token.lexeme;
+        int length = comment.length;
+        int leftIndex = comment.indexOf('[');
+        while (leftIndex >= 0 && leftIndex + 1 < length) {
```

Comment:
nit: I don't think it's necessary to test `leftIndex + 1 < length`. When that's `false`, `rightIndex` will be negative.

Reply:
Good catch.  Fixed!